telluric/georaster.py

```diff
@@ -1630,10 +1551,19 @@ def get_tile(self, x_tile, y_tile, zoom,
         :param blocksize: tile size  (x & y) default 256, for full resolution pass None
         :return: GeoRaster2 of tile
         """
-        coordinates = mercantile.xy_bounds(x_tile, y_tile, zoom)
-        window = self._window(coordinates)
-        return self.get_window(window, bands=bands,
-                               xsize=blocksize, ysize=blocksize)
+        roi = GeoVector.from_xyz(x_tile, y_tile, zoom)
+        coordinates = roi.get_bounds(WEB_MERCATOR_CRS)
+        window = self._window(coordinates, to_round=False)
+        window.col_off = round(window.col_off)
+        window.row_off = round(window.row_off)
+        window.width = round(window.width)
+        window.height = round(window.height)
+        bands = bands or list(range(1, self.num_bands + 1))
+        # we know the affine the result should produce becuase we know where
+        # it is located by the xyz, therefore we calculate it here
+        resolution = MERCATOR_RESOLUTION_MAPPING[zoom]
+        affine = Affine.translation(coordinates.left, coordinates.top) * Affine.scale(resolution, -resolution)
```

Comment:
this is not always right, here you are guessing, what you need to do is to `scale` the raster affine  (`self.affine`) to the new affine based on the ratio between the raster resolution and the tile resolution. regarding the translation, it is not always `left, top` it can be `left, bottom` depending of the direction of the scale. you can test this with a raster with positive `e` or negative `a`.